### PR TITLE
registry: reject new labels on deployment status updates

### PR DESCRIPTION
**Pull request by 0xmichalis**
@bgrant0607 @janetkuo PTAL

### CODE REVIEW

File: pkg/registry/deployment/strategy.go [hunk 1]
```diff
@@ -92,6 +92,8 @@ func (deploymentStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Spec = oldDeployment.Spec
+	newDeployment.Labels = oldDeployment.Labels
```
Comment (bgrant0607) [inline pkg/registry/deployment/strategy.go]: Do we do this for other APIs? We should be consistent, and document this in api conventions.

Does the deployment controller use resourceVersion to ensure it doesn't accidentally clobber labels and annotations?

I agree the controller shouldn't be able to update labels, but I could imagine cases where we'd want it to update annotations in the future.

Comment (janetkuo) [inline pkg/registry/deployment/strategy.go]: This strategy is for updating deployment status only (`UpdateStatus`). To update annotations/labels, `Update` should be called instead.

Comment (kargakis) [inline pkg/registry/deployment/strategy.go]: @bgrant0607 , @janetkuo is right, this is status update. I don't think it happens elsewhere.

Comment (bgrant0607) [inline pkg/registry/deployment/strategy.go]: @kargakis @janetkuo I understand this is status update.

We support updating annotations via the /binding subresource. There are use cases for controllers to update annotations along with status, though we don't do it now.

I agree that we shouldn't allow labels to be changed via status updates. @kargakis Please file an issue to fix the other status endpoints. It is technically a breaking API change, but one could view it as a vulnerability, so we should fix it.

Comment (kargakis) [inline pkg/registry/deployment/strategy.go]: opened https://github.com/kubernetes/kubernetes/issues/20978

Comment (bgrant0607) [inline pkg/registry/deployment/strategy.go]: Commented on that issue about why we may want to allow annotation updates.

**Comment by kargakis** [comment]
@bgrant0607 updated to reject only label updates. PTAL

**Comment by bgrant0607** [comment]
Lgtm

**Pull request merged**